tools/codegen/api/cpp.py

```diff
@@ -291,36 +267,22 @@ def pred(name: str, ty: Type) -> Callable[[Argument], bool]:
         pred('pin_memory', Type.parse('bool')),
     ]
 
-    has_tensoroptions_argument = False
     i = 0
     while i < len(func.kwarg_only_arguments):
         # If there is enough space...
         if i <= len(func.kwarg_only_arguments) - len(predicates):
             # And the next len(predicates) arguments look like TensorOptions arguments
             if all(p(a) for p, a in zip(predicates, func.kwarg_only_arguments[i : i + len(predicates)])):
-                has_tensoroptions_argument = True
                 # Group them together as one argument
-                gathered_args.append(TensorOptionsArguments(
+                args.append(TensorOptionsArguments(
```

Comment:
Not making a judgment call on if it's right or not, but I made it local because I believed there was only one use site, so might as well keep it close and encapsulated.

Reply:
I think my impulse here comes from a relentless compulsion to pull everything that's essentially part of the data model out of the code and into the global scope.